<?xml version='1.0'?>

<launch>
    <!-- Input Arguments -->
    <arg name="traj_topic_name" default="/offline_planner/trajectory" />
    <arg name="steering_topic_name" default="/SteeringAngle" />
    <arg name="throttle_topic_name" default="/cmd_vel" />
    <!-- <arg name="traj_time_scale_factor" default="1.025" /> -->
    <arg name="traj_time_scale_factor" default="1.0" />
    <arg name="velocity_max" default="33.33" />
    <arg name="lookahead" default="0.0" />
    <arg name="wheelbase" default="2.269" />
    <arg name="rear_to_cg" default="1.1345" />
    <arg name="sample_time" default="0.02" />

    <!-- Launch Node -->
    <node name="open_loop_controller" pkg="vehicle_control_pkg" type="open_loop_controller" output="screen">
        
        <!-- Input Parameters -->
        <param name="traj_topic_name" value="$(arg traj_topic_name)" />
        <param name="steering_topic_name" value="$(arg steering_topic_name)" />
        <param name="throttle_topic_name" value="$(arg throttle_topic_name)" />
        <param name="traj_time_scale_factor" value="$(arg traj_time_scale_factor)" />
        <param name="velocity_max" value="$(arg velocity_max)" />
        <param name="lookahead" value="$(arg lookahead)" />
        <param name="wheelbase" value="$(arg wheelbase)" />
        <param name="rear_to_cg" value="$(arg rear_to_cg)" />
        <param name="sample_time" value="$(arg sample_time)" />
    </node>

</launch>